src/controllers/controller.doughnut.js

```diff
@@ -36,16 +36,25 @@ module.exports = function(Chart) {
 			labels: {
 				generateLabels: function(data) {
 					if (data.labels.length && data.datasets.length) {
+
 						return data.labels.map(function(label, i) {
+							var ds = data.datasets[0];
+							var arc = ds.metaData[i];
```

Comment:
good catch. i don't know what the best way to solve it would be. I wonder if `buildOrUpdateElements()` can happen first